airbyte-cdk/python/airbyte_cdk/config_observation.py

```diff
@@ -68,10 +68,14 @@ def observe_connector_config(non_observed_connector_config: MutableMapping[str,
 
 
 def emit_configuration_as_airbyte_control_message(config: MutableMapping):
+    airbyte_message = create_connector_config_control_message(config)
+    print(airbyte_message.json(exclude_unset=True))
```

Comment:
As part of the declarative framework, we don't need to. However for compatibility reasons, we probably want to keep this function since maybe someone somewhere uses this

Reply:
Is this ever used outside of the context of an abstract source?

Reply:
So there are a couple of cases:
1. [The repository is optional right now](https://github.com/airbytehq/airbyte/pull/27017/files#diff-5db0a7e5cc64b411e940d1cbce5592aa44ab3a6573a13a98c7c021f490d93a3fR233). This is to keep backward compatibility. What I did omit is that there should be a `else: emit_configuration_as_airbyte_control_message(...)`
2. This solution does not cover other source methods. For a more detailed explanation, see the answer of the first question on https://github.com/airbytehq/airbyte/pull/27017#issuecomment-1578976209
3. I haven't done a full search of the usages of `ObservedDict` and `ConfigObserver`. At first, they seem unused but it feels weird to me that we have those but they aren't used

There might be other cases as well which I haven't thought about so don't hesitate to flag things you might find worrisome.